mathesar_ui/src/component-library/common/utils/contextValidationUtil.ts

```diff
@@ -0,0 +1,46 @@
+import { setContext, hasContext, getContext, onDestroy } from 'svelte';
+import { writable } from 'svelte/store';
+import type { Writable } from 'svelte/store';
+
+export type ValidationFunction = () => boolean;
+export type ValidationResultStore = Writable<boolean>;
+
+const VALIDATION_CONTEXT_KEY = 'validationContext';
+
+class ContextBasedValidator {
+  public validationResult: Writable<boolean> = writable(true);
```

Comment:
Agreed. The `ContextBasedValidator` class is currently only used on parent components to identify if all the child components are validated. The form validation logic is independent and contains the necessary information (Refer `FormValidationResult` interface).

I can see the value in having the error messages in `ContextBasedValidator` since it is a common utility of it's own and can be used directly.